File: backend/app/schemas/transcript.py
"""
Pydantic schemas for Transcript API
"""
from pydantic import BaseModel
from typing import Optional
from datetime import datetime
from uuid import UUID
from decimal import Decimal


class TranscriptResponse(BaseModel):
    """Response schema for transcript"""
    id: UUID
    call_id: UUID
    speaker: str
    text: str
    timestamp: datetime
    confidence: Optional[Decimal]
    created_at: datetime

    class Config:
        from_attributes = True


class TranscriptListResponse(BaseModel):
    """Response schema for list of transcripts"""
    call_id: UUID
    transcripts: list[TranscriptResponse]

File: frontend/src/types/transcript.ts
/**
 * Transcript-related type definitions
 */

export type SpeakerType = 'user' | 'ai';

export interface Transcript {
  id: string;
  call_id: string;
  speaker: SpeakerType;
  text: string;
  timestamp: string;
  confidence?: number;
  created_at: string;
}

export interface TranscriptListResponse {
  call_id: string;
  transcripts: Transcript[];
}

export interface WebSocketTranscriptMessage {
  type: 'transcript';
  speaker: SpeakerType;
  text: string;
  timestamp: string;
}
